src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1200,24 +1254,21 @@ private void decoderControl(Map<String, String> stats, List<AdvancedControllable
 				this.localDecoderInfoList.set(decoderID, decoderInfo);
 				populateDecoderControlBufferingMode(stats, advancedControllableProperties, decoderInfo);
 				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
-
 				populateLocalExtendedStats(stats, advancedControllableProperties);
 				break;
 			case BUFFERING_DELAY:
 				decoderInfo.setBufferingDelay(Integer.parseInt(value));
 				this.localDecoderInfoList.set(decoderID, decoderInfo);
-				advancedControllableProperties.add(createNumeric(stats, decoderControllingGroup + DecoderControllingMetric.BUFFERING_DELAY.getName(), decoderInfo.getBufferingDelay()));
+				advancedControllableProperties.add(createNumeric(stats, decoderControllingGroup + DecoderControllingMetric.BUFFERING_DELAY.getName(), decoderInfo.getBufferingDelay().toString()));
 				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
-
 				populateLocalExtendedStats(stats, advancedControllableProperties);
 				break;
 			case MULTI_SYNC_BUFFERING_DELAY:
 				decoderInfo.setMultisyncBufferingDelay(Integer.parseInt(value));
 				this.localDecoderInfoList.set(decoderID, decoderInfo);
 				advancedControllableProperties.add(
-						createNumeric(stats, decoderControllingGroup + DecoderControllingMetric.MULTI_SYNC_BUFFERING_DELAY.getName(), decoderInfo.getMultisyncBufferingDelay()));
+						createNumeric(stats, decoderControllingGroup + DecoderControllingMetric.MULTI_SYNC_BUFFERING_DELAY.getName(), decoderInfo.getMultisyncBufferingDelay().toString()));
```

Comment:
can getMultisyncBufferingDelay() be null?

Reply:
I fixed it, it can't be null any more